svtools/lmerge.py

```diff
@@ -240,280 +84,410 @@ def merge(BP, sample_order, v_id, use_product, include_genotypes=False):
+
+    var.set_info('SVTYPE', 'BND')
+    var.alt = ALT
+    
+    [ tempci, temp95, temppr ] = [ var.get_info('CIPOS'), var.get_info('CIPOS95'), var.get_info('PRPOS')]
+    var.set_info('CIPOS', var.get_info('CIEND'))
+    var.set_info('CIEND', tempci)
+    var.set_info('CIPOS95', var.get_info('CIEND95'))
+    var.set_info('CIEND95', temp95 )
+    var.set_info('PRPOS', var.get_info('PREND'))
+    var.set_info('PREND', temppr )
+
+
+def write_var(var, vcf_out, include_genotypes=False):
+
+    v_id=var.var_id
+    if var.get_info('CIPOS95') != '0,0' or var.get_info('CIEND95') != '0,0':
+        var.set_info('IMPRECISE', True)
+    else:
+        var.set_info('IMPRECISE', False)
+
+    if var.get_info('SVTYPE') == 'INV' and ('--:0' in var.get_info('STRANDS') or '++:0' in var.get_info('STRANDS')):
+
+        invtobnd(var)
+
+    if var.alt not in ['<DEL>', '<DUP>', '<INV>']:
+
+        var.var_id=str(v_id)+'_1'
+        var.set_info('EVENT', v_id)
+        var.set_info('MATEID', str(v_id)+'_2')
+        var.info.pop('END', None)
+        var.info.pop('SVLEN', None)
+        
+        varstring=var.get_var_string(use_cached_gt_string=True)
+        if not include_genotypes: 
+            varstring='\t'.join(varstring.split('\t', 10)[:9])
 
-        INFO = ';'.join(I)
+        vcf_out.write(varstring+'\n')
+        altstr=re.split( '[\[\]:]', var.alt)
 
-        QUAL = str(QUAL)
+        new_alt = ''
+        
+        if var.alt[0] == '[':
+            new_alt = '[' + var.chrom + ':' + str(var.pos) + '[N'
+        elif var.alt[0] == ']':
+            new_alt = 'N[' + var.chrom + ':' + str(var.pos) + '['
+        elif var.alt[-1] == '[':
+            new_alt = ']' + var.chrom + ':' + str(var.pos) + ']N'
+        elif var.alt[-1] == ']':
+            new_alt = 'N]' + var.chrom + ':' + str(var.pos) + ']'
 
-        O = [CHROM,POS,ID,REF,ALT,QUAL,FILTER,INFO]
 
-        print_var_line('\t'.join([str(o) for o in O]), GTS)
+        var.chrom=altstr[1]
+        var.pos=int(altstr[2])        
+        var.var_id=str(v_id)+'_2'
+        var.set_info('MATEID', str(v_id)+'_1')
+        var.set_info('SECONDARY', True)
+        var.alt=new_alt
+
+        [ tempci, temp95, temppr ] = [ var.get_info('CIPOS'), var.get_info('CIPOS95'), var.get_info('PRPOS')]
+        var.set_info('CIPOS', var.get_info('CIEND'))
+        var.set_info('CIEND', tempci)
+        var.set_info('CIPOS95', var.get_info('CIEND95'))
+        var.set_info('CIEND95', temp95 )
+        var.set_info('PRPOS', var.get_info('PREND'))
+        var.set_info('PREND', temppr )
+
+        varstring=var.get_var_string(use_cached_gt_string=True)
+        if not include_genotypes: 
+            varstring='\t'.join(varstring.split('\t', 10)[:9])
+
+        vcf_out.write(varstring+'\n')
+
+
+    else:
+        varstring=var.get_var_string(use_cached_gt_string=True)
+        if not include_genotypes: 
+            varstring='\t'.join(varstring.split('\t', 10)[:9])
```

Comment:
If we're not including genotypes then we probably don't want to include the format column. Looking at the current VCF code - the hacky way to do this would be to set var.format_dict to None. 

That's probably inappropriate though for the long term. I suspect the real change should be in the VCF classes. Perhaps by adding a test if the gts are None and if so, not printing out format.